### PR TITLE
Let mutating webhook defaults the object

### LINKED ISSUE #57982: Mutating Admission Webhooks: apiserver needs to default patched objects
Example: if you add an initContainer to a pod, you have to set fields that ought to have system-applied defaults.

It is not future-safe for webhook authors to supply their own defaults, therefore I believe we should fix this and backport it.

### PR DESCRIPTION
**Pull request by caesarxuchao**
...after applying the patch sent back by the webhook

This should be treated as a bug and cherrypicked to 1.9.

Fixes #57982

```release note
Mutating webhook controller now defaults the object after applying each patch sent back by the webhook.
```

### CODE REVIEW

**Comment by caesarxuchao** [comment]
@dashpole it seems that you are the branch manager, could you approve the milestone? It's fixing #57982. Thanks.

**Comment by dashpole** [comment]
@mbohlool is the patch manager for 1.9.

**Comment by caesarxuchao** [comment]
@mbohlool could you approve the milestone? It's fixing #57982. Thanks.

**Comment by lavalamp** [comment]
Thanks, Chao!

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go [hunk 1]
```diff
@@ -171,6 +173,9 @@ func (a *MutatingWebhook) ValidateInitialization() error {
 	if err := a.convertor.Validate(); err != nil {
 		return fmt.Errorf("MutatingWebhook.convertor is not properly setup: %v", err)
 	}
+	if a.defaulter == nil {
+		return fmt.Errorf("MutatingWebhook.defaulter is not properly setup: %v")
```
Comment (jennybuckley) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go]: What is the %v supposed to resolve to?

Comment (jennybuckley) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go]: I think it will say "%!v(MISSING)"

Comment (caesarxuchao) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go]: Thanks. Sent https://github.com/kubernetes/kubernetes/pull/58187.

**Pull request merged**